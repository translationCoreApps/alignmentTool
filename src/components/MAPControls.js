import React from 'react';
import PropTypes from 'prop-types';
import InfoIcon from 'material-ui/svg-icons/action/info';
import CheckIcon from 'material-ui/svg-icons/action/check-circle';
import CancelIcon from 'material-ui/svg-icons/navigation/cancel';
import RefreshIcon from 'material-ui/svg-icons/navigation/refresh';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

/**
 * Renders a secondary styled button
 * @param {*} [children] - the button content
 * @param {bool} [disabled=false] - controls whether the button is disabled
 * @param {func} [onClick] - optional click handler
 * @return {*}
 * @constructor
 */
const SecondaryButton = ({children, disabled, onClick}) => (
  <button className="btn-second"
          disabled={disabled}
          onClick={onClick}>
    {children}
  </button>
);
SecondaryButton.propTypes = {
  children: PropTypes.any,
  disabled: PropTypes.bool,
  onClick: PropTypes.func
};
SecondaryButton.defaultProps = {
  disabled: false
};

const styles = {
  root: {
    width: '100%',
    padding: '0 10px',
    boxShadow: 'rgba(0, 0, 0, 0.21) 0px -5px 5px 0px',
    textAlign: 'center'
  },
  button: {
    marginLeft: 10
  },
  icon: {
    color: 'var(--accent-color-dark)',
    verticalAlign: 'middle',
    marginRight: '5px',
    width: 30,
    height: 30
  },
  buttonIcon: {
    color: 'var(--accent-color-dark)',
    verticalAlign: 'middle',
    marginRight: '5px',

    width: 20,
    height: 20
  }
};

/**
 * Renders controls for managing Word MAP predictions
 * @param {func} onRefresh
 * @param {func} onAccept
 * @param {func} nReject
 */
class MAPControls extends React.Component {
  render() {
    const {onRefresh, onAccept, onReject, translate} = this.props;
    return (
      <MuiThemeProvider>
        <div style={styles.root}>
          <InfoIcon style={styles.icon}/>
          <SecondaryButton style={styles.button} onClick={onRefresh}>
            <RefreshIcon style={styles.buttonIcon}/>
            {translate('suggestions.refresh')}
          </SecondaryButton>
          <SecondaryButton style={styles.button} onClick={onAccept}>
            <CheckIcon style={styles.buttonIcon}/>
            {translate('suggestions.accept')}
          </SecondaryButton>
          <SecondaryButton style={styles.button} onClick={onReject}>
            <CancelIcon style={styles.buttonIcon}/>
            {translate('suggestions.reject')}
          </SecondaryButton>
        </div>
      </MuiThemeProvider>
    );
  }
}

MAPControls.propTypes = {
  onRefresh: PropTypes.func.isRequired,
  onAccept: PropTypes.func.isRequired,
  onReject: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};
export default MAPControls;
