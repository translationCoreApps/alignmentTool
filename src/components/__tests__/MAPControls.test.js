import MAPControls from '../MAPControls';
import React from 'react';
import renderer from 'react-test-renderer';

describe('MAPControls', () => {
  it('renders', () => {
    const wrapper = renderer.create(
      <MAPControls
        showPopover={jest.fn()}
        translate={k=>k}
        onAccept={jest.fn()}
        onReject={jest.fn()}
        onRefresh={jest.fn()}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
