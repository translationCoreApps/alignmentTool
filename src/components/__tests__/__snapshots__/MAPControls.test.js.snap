// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MAPControls renders 1`] = `
<MuiThemeProvider>
  <div
    style={
      Object {
        "boxShadow": "rgba(0, 0, 0, 0.21) 0px -5px 5px 0px",
        "padding": "0 10px",
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    <ActionInfo
      onClick={[Function]}
      style={
        Object {
          "color": "var(--accent-color-dark)",
          "cursor": "pointer",
          "height": 30,
          "marginRight": "5px",
          "verticalAlign": "middle",
          "width": 30,
        }
      }
    />
    <Tooltip
      delayHide={100}
      delayShow={1000}
      effect="solid"
      location="bottom"
      tooltip="suggestions.refresh_suggestions"
      type="dark"
    >
      <SecondaryButton
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <NavigationRefresh
          style={
            Object {
              "color": "var(--accent-color-dark)",
              "height": 20,
              "marginRight": "5px",
              "verticalAlign": "middle",
              "width": 20,
            }
          }
        />
        suggestions.refresh
      </SecondaryButton>
    </Tooltip>
    <Tooltip
      delayHide={100}
      delayShow={1000}
      effect="solid"
      location="bottom"
      tooltip="suggestions.accept_suggestions"
      type="dark"
    >
      <SecondaryButton
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <ActionCheckCircle
          style={
            Object {
              "color": "var(--accent-color-dark)",
              "height": 20,
              "marginRight": "5px",
              "verticalAlign": "middle",
              "width": 20,
            }
          }
        />
        suggestions.accept
      </SecondaryButton>
    </Tooltip>
    <Tooltip
      delayHide={100}
      delayShow={1000}
      effect="solid"
      location="bottom"
      tooltip="suggestions.reject_suggestions"
      type="dark"
    >
      <SecondaryButton
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "marginLeft": 10,
          }
        }
      >
        <NavigationCancel
          style={
            Object {
              "color": "var(--accent-color-dark)",
              "height": 20,
              "marginRight": "5px",
              "verticalAlign": "middle",
              "width": 20,
            }
          }
        />
        suggestions.reject
      </SecondaryButton>
    </Tooltip>
    <Toggle
      defaultToggled={false}
      disabled={false}
      iconStyle={
        Object {
          "marginTop": "8px",
        }
      }
      label="alignment_complete"
      labelPosition="right"
      labelStyle={
        Object {
          "color": "var(--accent-color-dark)",
          "lineHeight": "18px",
          "textAlign": "left",
        }
      }
      onToggle={[Function]}
      style={
        Object {
          "display": "inline-block",
          "fontSize": "14px",
          "margin": "10px",
          "verticalAlign": "middle",
          "width": "auto",
        }
      }
      thumbSwitchedStyle={
        Object {
          "backgroundColor": "var(--accent-color-dark)",
        }
      }
      toggled={false}
      trackSwitchedStyle={
        Object {
          "backgroundColor": "var(--accent-color)",
        }
      }
    />
  </div>
</MuiThemeProvider>
`;
